### PR TITLE
Fix school status when school user added via support

### PR DESCRIPTION
**Pull request by benilovj**
### Context

Before #560, the only way to add the first user to a school was to invite a school contact. It's now possible for support users to add the first user directly. The school status isn't being updated correctly in this case.

### Changes proposed in this pull request

Update the school status inference logic to correctly calculate the status, irrespective of how the first user was added.
Ensure that the school status is recalculated when a school user is added.

### CODE REVIEW

**Review by asmega — APPROVED** [review]
seems fine

File: app/models/preorder_information.rb [hunk 1]
```diff
@@ -42,6 +42,10 @@ def infer_status
     end
   end
 
+  def refresh_status
```
Comment (aldavidson) [inline app/models/preorder_information.rb:45]: minor niggle - can this method be `refresh_status!` as it both saves state, and can throw an exception?

File: app/models/preorder_information.rb [hunk 2]
```diff
@@ -27,13 +27,13 @@ def initialize(*args)
...
   def infer_status
-    if school_will_order_devices? && school_contact.nil?
+    if school_will_order_devices? && !any_school_users? && school_contact.nil?
       'needs_contact'
-    elsif school_will_order_devices? && school_contacted_at.nil?
+    elsif school_will_order_devices? && !any_school_users? && school_contact.present?
       'school_will_be_contacted'
-    elsif school_will_order_devices? && school_contacted_at.present? && !chromebook_information_complete?
+    elsif school_will_order_devices? && any_school_users? && !chromebook_information_complete?
       'school_contacted'
```
Comment (aldavidson) [inline app/models/preorder_information.rb:35]: If I'm reading this right, then we're no longer using the `school_contact` field to infer whether the school has been contacted or not. It's now primarily whether the school has any users or not. 

So is "school will be contacted" / "school_contacted" still the right set of names for these two states? (changing the names could of course be a separate PR)

**Review by aldavidson — APPROVED** [review]
No blockers, but one name-of-method question, and one slightly deeper question about the state logic

File: app/models/preorder_information.rb (hunk 1 above)
Comment (benilovj) [reply app/models/preorder_information.rb]: fair - done!

File: app/models/preorder_information.rb (hunk 2 above)
Comment (benilovj) [reply app/models/preorder_information.rb:35]: I think we will be able to retire the "school will be contacted" status – it's currently used to park:

- schools we've not contacted yet until the service can scale to handle the volume
- schools we can't contact yet because we can't add users to multiple schools yet

After we solve both, we should revisit the statuses and pare stuff down...
